### PR TITLE
[4.0.0-alpha.2] Backport changes from master

### PR DESCRIPTION
**Pull request by WiXSL**

### CODE REVIEW

File: packages/ra-ui-materialui/src/input/AutocompleteArrayInput.stories.tsx [hunk 1]
```diff
@@ -337,17 +337,16 @@ const CreateAuthor = () => {
     const { filter, onCancel, onCreate } = useCreateSuggestionContext();
     const [name, setName] = React.useState(filter || '');
     const [language, setLanguage] = React.useState('');
-    const [create] = useCreate('authors');
+    const [create] = useCreate();
 
     const handleSubmit = event => {
         event.preventDefault();
         create(
+            'authors',
```
Comment (WiXSL) [inline packages/ra-ui-materialui/src/input/AutocompleteArrayInput.stories.tsx:345]: Fix story

File: packages/ra-ui-materialui/src/form/Toolbar.tsx [hunk 2]
```diff
@@ -51,9 +51,7 @@ import { SaveButton, DeleteButton } from '../button';
  * @prop {string} width Apply to the mobile or desktop classes depending on its value. Pass `xs` to display the mobile version.
  *
  */
-export const Toolbar = <RecordType extends Partial<Record> = Partial<Record>>(
-    props: ToolbarProps<RecordType>
-) => {
+export const Toolbar: FC<ToolbarProps> = props => {
```
Comment (WiXSL) [inline packages/ra-ui-materialui/src/form/Toolbar.tsx:54]: Fix Toolbar def to avoid errors because of `ToolbarProps` definition change

Comment (fzaninotto) [reply packages/ra-ui-materialui/src/form/Toolbar.tsx:54]: I'd like to keep the toolbar Generic in next

File: packages/ra-ui-materialui/src/form/Toolbar.tsx [hunk 3]
```diff
@@ -175,7 +172,7 @@ export interface ToolbarProps<RecordType extends Partial<Record> = Record>
     submitOnEnter?: boolean;
     redirect?: RedirectionSideEffect;
     basePath?: string;
-    record?: RecordType;
+    record?: any;
```
Comment (fzaninotto) [inline packages/ra-ui-materialui/src/form/Toolbar.tsx:175]: please revert this one

File: packages/ra-ui-materialui/src/form/Toolbar.tsx (hunk 2 above)
Comment (WiXSL) [reply packages/ra-ui-materialui/src/form/Toolbar.tsx:54]: ok

File: packages/ra-ui-materialui/src/form/Toolbar.tsx (hunk 3 above)
Comment (WiXSL) [reply packages/ra-ui-materialui/src/form/Toolbar.tsx:175]: ok

File: packages/ra-ui-materialui/src/list/ListActions.tsx [hunk 4]
```diff
@@ -87,7 +87,19 @@ export const ListActions = (props: ListActionsProps) => {
                 )}
             </TopToolbar>
         ),
-        [resource, displayedFilters, filterValues, selectedIds, filters, total] // eslint-disable-line react-hooks/exhaustive-deps
```
Comment (fzaninotto) [inline packages/ra-ui-materialui/src/list/ListActions.tsx:90]: linter warning

**Review by djhi — APPROVED** [review]

**Review by fzaninotto — APPROVED** [review]

**Pull request merged**